drivers/SPI.h

```diff
@@ -179,75 +186,85 @@ class SPI : private NonCopyable<SPI> {
         return 0;
     }
 
-    /** Abort the on-going SPI transfer, and continue with transfer's in the queue if any.
+    /** Abort the on-going SPI transfer, and continue with transfers in the queue if any.
      */
     void abort_transfer();
 
-    /** Clear the transaction buffer
+    /** Clear the queue of transfers.
      */
     void clear_transfer_buffer();
 
-    /** Clear the transaction buffer and abort on-going transfer.
+    /** Clear the queue of transfers and abort the on-going transfer.
      */
     void abort_all_transfers();
 
-    /** Configure DMA usage suggestion for non-blocking transfers
+    /** Configure DMA usage suggestion for non-blocking transfers.
+     *
+     *  @param usage The usage DMA hint for peripheral.
      *
-     *  @param usage The usage DMA hint for peripheral
-     *  @return Zero if the usage was set, -1 if a transaction is on-going
-    */
+     *  @return Result of the operation.
+     *  @retval 0 The usage was set.
+     *  @retval -1 Usage cannot be set as there is an ongoing transaction.
+     */
     int set_dma_usage(DMAUsage usage);
 
 protected:
-    /** SPI IRQ handler
-     *
-    */
+    /** SPI interrupt handler.
+     */
     void irq_handler_asynch(void);
 
-    /** Common transfer method
+    /** Start the transfer or put it on the queue.
      *
      * @param tx_buffer The TX buffer with data to be transfered. If NULL is passed,
      *                  the default SPI value is sent
-     * @param tx_length The length of TX buffer in bytes
+     * @param tx_length The length of TX buffer in bytes.
      * @param rx_buffer The RX buffer which is used for received data. If NULL is passed,
-     *                  received data are ignored
-     * @param rx_length The length of RX buffer in bytes
-     * @param bit_width The buffers element width
-     * @param callback  The event callback function
-     * @param event     The logical OR of events to modify
-     * @return Zero if the transfer has started or was added to the queue, or -1 if SPI peripheral is busy/buffer is full
-    */
+     *                  received data are ignored.
+     * @param rx_length The length of RX buffer in bytes.
+     * @param bit_width The buffers element width in bits.
+     * @param callback  The event callback function.
+     * @param event     The event mask of events to modify.
+     *
+     * @return Operation success.
+     * @retval 0 A transfer was started or added to the queue.
+     * @retval -1 Transfer can't be added because queue is full.
```

Comment:
there is actually a bug in the implementation, it doesn't forward the error code, will file an issue